### ISSUE TITLE
add congratulations letter to dashboard

### ISSUE THREAD

**Issue by pdpinch** [mitodl/micromasters#4247]
As a learner who has earned a program certificate, I sometimes have trouble explaining the meaning and value of the certificate. It's useful to have a letter from the faculty that describes my accomplishment that I can share with employers, in addition to the certificate. 

### acceptance criteria:
- [ ] SCM only, to start
  copy, logos and signatures will have to be customized in the future
- [ ] behind a feature flag
- [ ] Only available to learners who have earned program credential
- [ ] Link on dashboard, below the link to the program 
- [ ] page prints reasonably well
- [ ] URL is shareable with users without MM accounts (similar to certificate and program certificate)

### questions
**Q:** Can we allow the course team to manage the copy through the CMS?
**A:** with preview not working in the CMS, this is of limited value. Also, after the initial creation, it may not need to be updated very frequently. 
**Q:** Signature images and titles for certificates are already handled in the CMS. Can we do the same thing for the letter signatures?

**Comment by pdpinch** [comment]
Draft of the letter: [MITx MicroMasters in SCM Letter of Accomplishment .pdf](https://github.com/mitodl/micromasters/files/2832488/MITx.MicroMasters.in.SCM.Letter.of.Accomplishment.pdf)

**Comment by pdpinch** [comment]
@annagav can you take a look at this and propose an approach? I have some thoughts if you want to talk about it first.

**Comment by annagav** [comment]
@pdpinch SCM program as a non-FA program does not have MicromastersProgramCertificate created after completion of the program. So there is no link to program certificate. It just looks like this currently: 
<img width="438" alt="screen shot 2019-02-07 at 11 57 09 am" src="https://user-images.githubusercontent.com/7574259/52428240-92bd7f00-2acf-11e9-8dc1-9cefe201ec71.png">

**Comment by pdpinch** [comment]
Oh, I forgot about that. We should add a link at some point, but that's out of scope for this issue. (IIRC, the program certificate doesn't show up in the edX certificates API) 

We can still add a link here to "Letter of Accomplishment"

**Comment by pdpinch** [comment]
Anna I spoke, and made some preliminary decision. I'll have to update the description and probably make some sub-issues before we can start:

- We can't attach this to the program certificate, because SCM doesn't have program certificates in this app & we can't easily get them from edX. 
- We'll try using wagtail, since that seems like an easier path to customizing for each program than creating custom multiple custom templates.

**Comment by annagav** [comment]
I think this roughly the guideline: 

1. Create a model `MicromastersProgramCompletion` that has a unique hash, very similar to `ProgramCertificate`.
2. Add a scheduled task that finds all users that completed all courses in the program (only non-fa for now, but this might change in the future) and create ``MicromastersProgramCompletion`.
3. Create a view for the letter. Just like with `MicromastersProgramCertificate` some of the info should come from ProgramPage in waigtail (text and signatures).

**Comment by pdpinch** [comment]
This issue requires an RFC that fleshes out https://github.com/mitodl/micromasters/issues/4247#issuecomment-461533035 from Anna.

**Comment by tasawernawaz** [comment]
@pdpinch Can you explain what are the SCM and non-fa learners ?

@annagav Instead of creating hash manually, can we add auto generated uuid field (like edX use in credentials service)

**Comment by pdpinch** [comment]
`SCM` = Supply Chain Management, our first of four MicroMasters. You can see more about them at https://micromasters.mit.edu/scm

`fa` = financial Aid = micromasters programs that use personal pricing and our ecommerce code. The only fa MicroMasters is DEDP. Because learners don’t pay for these courses on edx.org, we rely on our micromasters app to create their course certificates and program certificates. 

`non-fa` = micromasters programs that don’t use personal pricing.
